snapcraft/internal/project_loader.py

```diff
@@ -133,13 +134,16 @@ def __init__(self, project_options=None):
         self.data = self._expand_env(snapcraft_yaml)
 
         # Prevent multiple parts from having duplicate alias names
-        aliases = []
-        for app in self.data.get('apps', []):
-            app = self.data['apps'][app]
-            for alias in app.get('aliases', []):
-                if alias in aliases:
-                    raise errors.DuplicateAliasError(alias=alias)
-                aliases.append(alias)
+        aliases = [
+            app.get('aliases', []) for app_name, app in self.data.get(
+                'apps', {}).items()]
+        aliases = list(itertools.chain.from_iterable(aliases))
+        unique_aliases = set(aliases)
+        if len(aliases) != len(unique_aliases):
+            for alias in unique_aliases:
+                aliases.remove(alias)
```

Comment:
Another annoying comment: This seems to be O(n^2), as you're traversing `unique_aliases` as well as `aliases` (remove() is O(n)).

```python
seen = set()
duplicated = set()
for alias in aliases:
    if alias in seen:
        duplicated.add(alias)
    else:
        seen.add(alias)
```

Seems like O(n), as `set().add()` is O(1) and `x in set()` is also O(1).